pkg/k8s/util.go

```diff
@@ -178,3 +186,79 @@ func NamespacedNameFrom(name string) (types.NamespacedName, error) {
 func IsHeadlessService(svc corev1.Service) bool {
 	return len(svc.Spec.ClusterIP) == 0 || svc.Spec.ClusterIP == corev1.ClusterIPNone
 }
+
+func StopJobSidecars(stop <-chan struct{}, broker *messaging.Broker, config *rest.Config, clientset kubernetes.Interface) {
```

Comment:
nit: we should avoid passing client objects as params when we're not doing so for dependency injection.

Instead, we should maintain a higher level client object (k8s.Controller, or something similar), and place this as a method on that type, which has access to the underlying (dependency injected client).

This maintains a cleaner approach for decoupling from API's and leveraging interfaces.

See https://docs.google.com/document/d/1L_u-XlVU8Fog-uFS3Uf0O2nz8IArQy82cd6Npxm038Y/edit#heading=h.cvqx4tbzt6i7